### ISSUE TITLE
Refactor to avoid unnecessary if statements

### ISSUE THREAD

**Issue by roice3** [mtwoodard/hypVR-Ray#43]
In globalsInclude.glsl, the geometryNormalize, geometryDot, etc. functions have if statements that check the geometry. I think it would be better to only declare the function signatures in this file and have the individual geometry files implement the functions, avoiding these if branches. Info on declaring is here:

https://relativity.net.au/gaming/glsl/Functions.html

Somewhat related, we can move some functions like sphereHSDF out of the specific geometry files because now that there is a geometryDistance function, the implementation is the same in every case. This should live in globalsInclude.glsl

In short, we should keep any geometry-specific code in the geometry files and any geometry-agnostic code in the globals file.
